lib/jxl/decode.cc

```diff
@@ -2413,15 +2433,22 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderImageOutBufferSize(
   if (format->num_channels < 3 && !dec->metadata.m.color_encoding.IsGray()) {
     return JXL_API_ERROR("Grayscale output not possible for color image");
   }
-
-  size_t row_size =
-      jxl::DivCeil(dec->metadata.oriented_xsize(dec->keep_orientation) *
-                       format->num_channels * bits,
-                   jxl::kBitsPerByte);
+  size_t oriented_xsize = dec->metadata.oriented_xsize(dec->keep_orientation);
```

Comment:
The comment in the header of this function could be updated to say that in case of no coalescing, it can only be called after the frame event (as opposed to after basic info as it says now), and will return the size required for the possibly cropped frame in that case, rather than the full image

Reply:
Added a comment.